daemon/logger/journald/read.go

```diff
@@ -237,7 +237,10 @@ drain:
 
 	// free(NULL) is safe
 	C.free(unsafe.Pointer(oldCursor))
-	C.sd_journal_get_cursor(j, &cursor)
+	if C.sd_journal_get_cursor(j, &cursor) != 0 {
+		// ensure that we won't be freeing an address that's invalid
```

Comment:
@tiborvass the man page for `sd_journal_get_cursor()` doesn't mention what happens to the cursor pointer when the function returns a failure code.  The implementation ends up calling `asprintf()`, which explicitly says the pointer's content is unspecified, so after a failure code we have to take a step to make sure that the pointer's not still uninitialized, and that it hasn't been set to some garbage value.